baseband/asp/payload.py

```diff
@@ -0,0 +1,51 @@
+import numpy as np
+from ..vlbi_base.payload import VLBIPayloadBase
+from collections import namedtuple
+
+
+__all__ = ['ASPPayload']
+
+
+def decode_8bit(words):
+    return words.view(np.int8, np.ndarray).astype(np.float32)
+
+
+def encode_8bit(values):
+    return np.clip(np.rint(values), -128, 127).astype(np.int8)
+
+
+NPOL = 2
+NDIM = 2
```

Comment:
As far as I can tell these are also hardcoded into dspsr, but the sample file's file header gives nchan = 8, as does `/mnt/scratch-lustre/mahajan/Ar_P2067/P2067_1/p2067/node16/2006_05_18_01/000007.asp.raw`.  Are the file headers not to be trusted?